reVX/wind_dirs/prominent_wind_dirs.py

```diff
@@ -0,0 +1,244 @@
+        """
+        Map powerrose directions to sc row and column shifts
+
+        Parameters
+        ----------
+        power_rose_h5_fpath : str
+            Filepath to .h5 file containing powerrose
+
+        Returns
+        -------
+        list
+            Pos of major cardinal directions in power rose data
+        """
+        directions = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
+
+        with h5py.File(power_rose_h5_fpath, 'r') as f:
+            cardinal_dirs = list(f['cardinal_directions'][...].astype(str))
+
+        dir_pos = [cardinal_dirs.index(d) for d in directions]
+
+        return dir_pos
+
+    @staticmethod
+    def _compute_neighbors(sc_point_gids, points):
+        """
+        Compute neighboring supply curve point gids in following order:
+        ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
+
+        Parameters
+        ----------
+        sc_point_gids : list | ndarray
+            Supply curve point gids to get neighbors for
+        points : pandas.DataFrame
+            Supply curve point gid to row, col index mapping
+
+        Returns
+        -------
+        neighbor_gids : ndarray
+            Neighboring supply curve point gids
+        """
+        sc_points = points.loc[sc_point_gids]
+        rows, cols = sc_points.values.T
+        shape = points['row_ind'].max(), points['col_ind'].max()
+
+        row_shifts = [-1, -1, 0, 1, 1, 1, 0, -1]
+
+        rows = np.expand_dims(rows, axis=1) + row_shifts
+        mask = rows < 0
+        rows[mask] = 0
+        mask = rows > shape[0]
+        rows[mask] = shape[0]
+
+        cols = np.expand_dims(cols, axis=1) + row_shifts
+        mask = cols < 0
+        cols[mask] = 0
+        mask = cols > shape[1]
+        cols[mask] = shape[1]
+
+        neighbor_gids = rows * shape[1] + cols
+
+        return neighbor_gids
+
+    @staticmethod
+    def _get_neighbors(excl_fpath, sc_point_gids, resolution=64):
+        """
+        Get neighboring sc_point_gids for all given supply curve points
+
+        Parameters
+        ----------
+        excl_fpath : str
+            Filepath to exclusions h5 with techmap dataset.
+        sc_point_gids : list | ndarray
+            List of supply curve point gids of interest
+        resolution : int, optional
+            SC resolution, must be input in combination with gid. Prefered
+            option is to use the row/col slices to define the SC point instead,
+            by default 64
+
+        Returns
+        -------
+        neighbor_gids : pandas.DataFrame
+            Neighboring sc_point_gids by cardinal direction
+        """
+        with SupplyCurveExtent(excl_fpath, resolution=resolution) as sc:
+            points = sc.points
+
+        neighbor_gids = \
+            ProminentWindDirections._compute_neighbors(sc_point_gids,
+                                                       points)
+
+        directions = ['N', 'NE', 'E', 'SE', 'S', 'SW', 'W', 'NW']
```

Comment:
This might be a good class attribute so people could easily see what the global direction order is.

Reply:
Done